src/soep_preparation/clean_variables/pgen.py

```diff
@@ -0,0 +1,220 @@
+    out["refugee_status"] = object_to_str_categorical(raw_data["pgstatus_refu"])
+    out["marital_status"] = object_to_str_categorical(raw_data["pgfamstd"])
+    out["laborforce_status"] = object_to_str_categorical(raw_data["pglfs"])
+    out["occupation_status"] = object_to_str_categorical(raw_data["pgstib"])
+    out["employment_status"] = object_to_str_categorical(raw_data["pgemplst"])
+    out["total_full_time_working_experience"] = object_to_float(raw_data["pgexpft"])
+    out["total_part_time_working_experience"] = object_to_float(raw_data["pgexppt"])
+    out["total_unemployment_experience"] = object_to_float(raw_data["pgexpue"])
+    out["tenure"] = object_to_float(raw_data["pgerwzeit"])
+    out["retired"] = create_dummy(out["occupation_status"], "NE: Rentner/Rentnerin")
+    out["in_education"] = _in_education(
+        out["employment_status"],
+        out["occupation_status"],
+    )
+    out["self_employed"] = create_dummy(
+        out["occupation_status"],
+        _self_employed_occupations(out["occupation_status"]),
+        "isin",
+    )
+    out["military"] = create_dummy(
+        out["occupation_status"],
+        "NE: Wehr- und Zivildienst",
+    )
+
+    out["erwerbstätig_y"] = (
+        create_dummy(out["employment_status"], "Nicht erwerbstätig", "neq")
+    ) & (~out["in_education"])
+
+    out["nicht_erwerbstätig_y"] = create_dummy(
+        out["employment_status"],
+        "Nicht erwerbstätig",
+    )
+    out["unemployed_y"] = create_dummy(
+        out["occupation_status"],
+        "NE: arbeitslos gemeldet",
+    )
+    out["ft_y"] = create_dummy(out["employment_status"], "Voll erwerbstätig")
+    out["pt_employed_y"] = create_dummy(
+        out["employment_status"], "Teilzeitbeschäftigung"
+    )
+    out["geringfügig_erwbstätig"] = create_dummy(
+        out["employment_status"],
+        "Unregelmässig, geringfügig erwerbstät.",
+    )
+    out["werkstatt"] = create_dummy(
+        out["employment_status"],
+        "Werkstatt für behinderte Menschen (seit 1998)",
+    )
+    out["beamte"] = create_dummy(
+        series=out["occupation_status"],
+        value_for_comparison="Beamte",
+        comparison_type="startswith",
+    )
+    out["parental_leave"] = create_dummy(
+        out["laborforce_status"],
+        "NE: Mutterschutz/Elternzeit (seit 1991)",
+    )
+
+    # individual work information
+    out["gross_labor_income_previous_month"] = object_to_float(
+        raw_data["pglabgro"]
+    ).fillna(0)
+    out["net_labor_income_previous_month"] = object_to_float(
+        raw_data["pglabnet"]
+    ).fillna(0)
+    out["weekly_working_hours_actual"] = _weekly_working_hours_fill_non_working(
+        raw_data["pgtatzeit"],
+        out["employment_status"],
+    )
+    out["weekly_working_hours_contract"] = _weekly_working_hours_fill_non_working(
+        raw_data["pgvebzeit"],
+        out["employment_status"],
+    )
+    out["public_service"] = object_to_bool_categorical(
```

Comment:
```suggestion
    out["employed_in_public_service"] = object_to_bool_categorical(
```

Too easy to confuse with "civil servant" (Beamter) otherwise.